content/guides/caching.md

```diff
@@ -4,284 +4,358 @@ sort: 11
+|- package.json
+|- webpack.config.js
+|- /dist
+|- /src
+  |- index.js
+|- /node_modules
 ```
 
-It is even easier to do with webpack. Each webpack build generates a unique hash which can be used to compose a filename, by including output [substitutions](/configuration/output#output-filename). The following example config will generate 2 files (1 per entry) with hashes in filenames:
-
-```js
-// webpack.config.js
-const path = require("path");
-
-module.exports = {
-  entry: {
-    vendor: "./src/vendor.js",
-    main: "./src/index.js"
-  },
-  output: {
-    path: path.join(__dirname, "build"),
-    filename: "[name].[hash].js"
-  }
-};
+__webpack.config.js__
+
+``` diff
+  const path = require('path');
+  const CleanWebpackPlugin = require('clean-webpack-plugin');
+  const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+  module.exports = {
+    entry: './src/index.js',
+    plugins: [
+      new CleanWebpackPlugin(['dist']),
+      new HtmlWebpackPlugin({
+-       title: 'Output Management'
++       title: 'Caching'
+      })
+    ],
+    output: {
+-     filename: 'bundle.js',
++     filename: '[name].[chunkhash].js',
+      path: path.resolve(__dirname, 'dist')
+    }
+  };
 ```
 
-Running webpack with this config will produce the following output:
-
-```bash
-Hash: 2a6c1fee4b5b0d2c9285
-Version: webpack 2.2.0
-Time: 62ms
-                         Asset     Size  Chunks             Chunk Names
-vendor.2a6c1fee4b5b0d2c9285.js  2.58 kB       0  [emitted]  vendor
-  main.2a6c1fee4b5b0d2c9285.js  2.57 kB       1  [emitted]  main
-   [0] ./src/index.js 63 bytes {1} [built]
-   [1] ./src/vendor.js 63 bytes {0} [built]
-```
+Running our build script, `npm run build`, with this configuration should produce the following output:
 
-But the problem here is, builds after *any file update* will update all filenames and clients will have to re-download all application code. So how can we guarantee that clients always get the latest versions of assets without re-downloading all of them?
+?> Add bash output
 
+Now our bundle's name should change to reflect its content, meaning if `src/index.js` is changed, the filename should change. However, if we run another build, we'll see that this is not the case:
 
-## Generating unique hashes for each file
+?> Add bash output
 
-What if we could produce the same filename, if the contents of the file did not change between builds? For example, it would be unnecessary to re-download a vendor file, when no dependencies have been updated, only application code.
+This is because webpack includes certain boilerplate, specifically the runtime and manifest, in the entry chunk.
 
-webpack allows you to generate hashes depending on file contents, by replacing the placeholder `[hash]` with `[chunkhash]`. Here is the updated config:
 
-```diff
-module.exports = {
-  /*...*/
-  output: {
-    /*...*/
--   filename: "[name].[hash].js"
-+   filename: "[name].[chunkhash].js"
-  }
-};
-```
+## Extracting Boilerplate
```

Comment:
Idk I think I'd rather leave this title as is for now. Partly because the mix of lowercase and uppercase (I know this is defined in our style guide/media repo) feels a bit odd but also I think it's pretty clear based on context.